providers/sparkpost/src/lib/sparkpost.error.ts

```diff
@@ -0,0 +1,16 @@
+export interface ISparkPostErrorResponse {
+  errors: Array<{
+    description: string;
+    code: string;
+    message: string;
+  }>;
+}
+
+export class SparkPostError extends Error implements ISparkPostErrorResponse {
+  readonly errors: ISparkPostErrorResponse['errors'];
+
+  constructor(response: ISparkPostErrorResponse, readonly statusCode: number) {
+    super();
+    this.errors = response.errors;
+  }
+}
```

Comment:
Why I try to create SparkPostError instead just throwing axios exception?

Because AxiosError is circular object (refers to itself in some field) and worker throws error when trying to stringify object like that.

I want to make sure that user will get helpful execution details.

Reply:
Oh really nice catch! I was wondering what was causing those errors with some providers